src/full_spectrum/neo4clj/client.clj

```diff
@@ -0,0 +1,118 @@
+      (neo4j/connect url usr pwd)
+      (neo4j/connect url usr pwd (neo4j/build-config opts)))))
+
+(defn disconnect
+  "Disconnect the given connection"
+  [^StatementRunner conn]
+  (neo4j/disconnect conn))
+
+(defn execute-multiple!
+  "Execute the given queries on the specified connection within the same session"
+  [conn & queries]
+  (with-open [session (neo4j/create-session conn)]
+    (for [query queries]
+      (let [query-string (if (string? query) query (first query))
+            params (if (coll? query) (second query) {})]
+        (neo4j/execute session query-string params)))))
+
+(defn execute!
+  "Execute the given query on the specified connection"
+  ([^StatementRunner conn ^String query]
+   (execute! conn query {}))
+  ([^StatementRunner conn ^String query ^clojure.lang.IPersistentMap params]
+   (with-open [session (neo4j/create-session conn)]
+     (neo4j/execute conn query params))))
+
+(defn execute-in-transaction
+  "Execute the given queries on the specified connection in a transaction"
+  [^StatementRunner conn & queries]
+  (with-open [trans (neo4j/begin-transaction (neo4j/create-session conn))]
+    (try
+      (let [result (execute-multiple! trans queries)]
+        (neo4j/commit-transaction trans)
+        result)
+      (catch Exception e
+        (neo4j/rollback-transaction trans)
+        (throw e)))))
+
+(defmacro execute-in-transaction2
+  [^StatementRunner conn trans & body]
+   `(with-open [~trans (neo4j/begin-transaction (neo4j/create-session ~conn))]
+     ~@body))
+
+(defn create-index!
+  [^StatementRunner conn label prop-key]
+  (execute! conn (builder/index-query "CREATE" label prop-key)))
+
+(defn drop-index!
+  [^StatementRunner conn label prop-key]
+  (execute! conn (builder/index-query "DROP" label prop-key)))
+
+(defn create-node!
+  [^StatementRunner conn ^clojure.lang.APersistentMap node]
+  (get (first (execute! conn (builder/create-node-query node true)))
+       (:ref-id node)))
+
+(defn find-nodes!
+  [^StatementRunner conn ^clojure.lang.APersistentMap node]
+  (map #(get % (:ref-id node))
+       (execute! conn (builder/lookup-query node true))))
+
+(defn create-relationship!
+  [^StatementRunner conn ^clojure.lang.APersistentMap rel]
+  (execute! conn (builder/create-relationship-query rel true)))
+
+(defn create-graph!
+  "Optimized function to create a whole graph within a transaction
+
+  Format of the graph is:
+  :lookups - collection of neo4j lookup representations
+  :nodes - collection of neo4j node representations
+  :relationships  - collection of neo4j relationship representations
+  :return-aliases - collection of aliases to return from query"
+  [^StatementRunner conn ^clojure.lang.APersistentMap graph]
+  (execute! conn (builder/create-graph-query graph)))
+
+(defn add-labels!
+  "Takes a collection of labels and adds them to the found neo4j entities"
+  [^StatementRunner conn ^clojure.lang.APersistentMap neo4j-entity ^clojure.lang.APersistentVector labels]
```

Comment:
Consider if it increases readability putting these on multiple lines for all functions with "long" argument declerations:
```clojure
[^StatementRunner conn
 ^clojure.lang.APersistentMap neo4j-entity
 ^clojure.lang.APersistentVector labels]
```